src/Functions/formatDateTime.cpp

```diff
@@ -943,7 +943,15 @@ class FunctionFormatDateTimeImpl : public IFunction
         {
             if constexpr (std::is_same_v<DataType, DataTypeDateTime64>)
             {
-                const auto c = DecimalUtils::split(vec[i], scale);
+                auto c = DecimalUtils::split(vec[i], scale);
+
+                if (vec[i].value < 0 && c.fractional)
```

Comment:
Probably it also good to add a comment about this behavior

Reply:
Added (added also comment to DecimalUtils::split() in [other pull request](https://github.com/ClickHouse/ClickHouse/pull/51291/files#diff-176a35d6aab5cab74176aa902f1519abcb359e38e73da9eb0d5c9b40703d0840L50))